core/iwasm/aot/debug/elf_parser.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2021 Ant Group.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <stdio.h>
+#include <assert.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+#include <stdbool.h>
+
+#include <elf.h>
+
+#include "aot_runtime.h"
+#include "bh_log.h"
+#include "elf_parser.h"
+
+
+bool
+is_ELF(void *buf)
+{
+    Elf32_Ehdr *eh = (Elf32_Ehdr *)buf;
+    if (!strncmp((char *)eh->e_ident, "\177ELF", 4)) {
+        LOG_VERBOSE("the buffer is ELF entry!");
+        return true;
+    }
+    LOG_VERBOSE("the buffer is not ELF entry!");
+    return false;
+}
+
+bool
+is64Bit(Elf32_Ehdr *eh)
+{
+    if (eh->e_ident[EI_CLASS] == ELFCLASS64)
+        return true;
+    else
+        return false;
+}
+
+bool
+is32Bit(Elf32_Ehdr *eh)
+{
+    if (eh->e_ident[EI_CLASS] == ELFCLASS32)
+        return true;
+    else
+        return false;
+}
+
+bool
+is_ELF64(void *buf)
+{
+    Elf64_Ehdr *eh = (Elf64_Ehdr *)buf;
+    if (!strncmp((char *)eh->e_ident, "\177ELF", 4)) {
+        LOG_VERBOSE("the buffer is ELF entry!");
+        return true;
+    }
+    LOG_VERBOSE("the buffer is not ELF entry!");
+    return false;
+}
+
+
+
+void
+read_section_header_table(Elf32_Ehdr *eh, Elf32_Shdr *sh_table[])
+{
+    uint32_t i;
+    char *buf = (char *)eh;
+    buf += eh->e_shoff;
+    LOG_VERBOSE("str index = %d count=%d", eh->e_shstrndx, eh->e_shnum);
+    for (i = 0; i < eh->e_shnum; i++) {
+        sh_table[i] = (Elf32_Shdr *)buf;
+        buf += eh->e_shentsize;
+    }
+}
+
+void
+read_section_header_table64(Elf64_Ehdr *eh, Elf64_Shdr *sh_table[])
+{
+    uint32_t i;
+    char *buf = (char *)eh;
+    buf += eh->e_shoff;
+
+    for (i = 0; i < eh->e_shnum; i++) {
+        sh_table[i] = (Elf64_Shdr *)buf;
+        buf += eh->e_shentsize;
+    }
+}
+
+char *
+get_section(Elf32_Ehdr *eh, Elf32_Shdr *section_header)
+{
+    char * buf = (char *)eh;
+    return buf + section_header->sh_offset;
+}
+
+char *
+get_section64(Elf64_Ehdr *eh, Elf64_Shdr *section_header)
+{
+    char * buf = (char *)eh;
+    return buf + section_header->sh_offset;
+}
+
+bool
+get_text_section(void *buf, uint64_t *offset, uint64_t *size)
+{
+    bool ret = false;
+    uint32_t i;
+    char* sh_str;
+
+    if (is64Bit(buf)) {
+        Elf64_Ehdr *eh = (Elf64_Ehdr *)buf;
+        Elf64_Shdr **sh_table = wasm_runtime_malloc(eh->e_shnum * sizeof(Elf64_Shdr *));
+        if(sh_table) {
```

Comment:
if (sh_table) but not if(sh_table)

Should handle return value when sh_table == NULL, e.g.,
```C
if (sh_table == NULL) {
    ...
    return false;
}